src/app/components/case/tasks/CamundaTask/utils/mapSubmitData.ts

```diff
@@ -0,0 +1,21 @@
+import { appendTimeToDate } from "app/components/shared/Helpers/helpers"
+
+type Rec = Record<string, { value: string | number | boolean }>
+
+const castValue = (item: any, value: any) => {
+  if (item.type === "select") return value.value
+  if (item.type === "checkbox") return value !== ""
+  if (item.camunda_type === "Long") return parseFloat(value)
+  if (item.is_date) return appendTimeToDate(value)
+  return value
+}
+
+export default (form: Components.Schemas.CamundaTask["form"], data: any) => {
+  const o = form.reduce((acc: Rec, item: any) => {
```

Comment:
I wouldn't mind for readable variable names

Reply:
Agree. Will improve.